### DISCUSSION TITLE
Mimic WPT experiments

### DISCUSSION THREAD

**Discussion by ghisleouf** [GoogleChrome/lighthouse#14945] [General]
Hi there,

I'm sending this message because I would be interested in knowing if there is a way to mimic "kind of" WPT experiments features.
Actually, my goal is quite simple to explain : I would like to re-run a Lighthouse test with modified DOM (removing tags, adding new one, ...).
I have no idea where to start and if there a way to achieve it.

If you have any idea, I would be pleased to listen to your advices.

Thanks a lot for your amazing work.

Best.

**Comment by adamraine** [comment]
Not really. Lighthouse won't modify the page for you and test the results. However, Lighthouse does *estimate* the impact of some suggested changes in the "Opportunities" section.

**Reply by ghisleouf** [reply]
Thanks for your answer.
What I had in mind was to find a way to hook the received HTML during LH test and replacing the real one by the HTML you want (that you could pass as a parameter to your test). That's the easy way.
My point is that I don't understand how WPT could achieve their experiments feature.

If someone has an idea, that would be awesome! I've found this issue on github : https://github.com/GoogleChrome/lighthouse/issues/14561 which is, I think, what I would like to do !

Thanks for your help ! :)

**Comment by connorjclark** [comment]
You can utilize [local overrides](https://www.debugbear.com/blog/devtools-local-overrides) in DevTools, which the Lighthouse panel will respect.
